qemu/tests/qemu_guest_agent.py

```diff
@@ -1580,14 +1605,14 @@ def gagent_check_file_read(self, test, params, env):
         # if guest os resource is enough, will return no error.
         # else it will return error like "insufficient system resource"
         # which is expected
-        try:
-            self.gagent.guest_file_read(ret_handle, count=1000000000)
-        except guest_agent.VAgentCmdError as detail:
-            info_insuffi = "Insufficient system resources exist to"
-            info_insuffi += " complete the requested service"
-            if not re.search(info_insuffi, str(detail)):
-                test.fail("Return error but is not the desired information: "
-                          "('%s')" % str(detail))
+        if params.get("os_type") == "linux":
+            main_qga_ver = self._get_main_qga_version(session, self.vm)
+        if params.get("os_type") == "linux" and main_qga_ver < 5:
+            _read_guest_file_with_count(1000000000)
+        else:
+            # For linux, if qga version higher than 5.0,the count
+            # number is restricted to below 48MB to avoid crashes.
+            _read_guest_file_with_count(10000000)
```

Comment:
And I recommend here:
```
count = 1000000000 if main_qga_ver < 5 else 10000000
_read_guest_file_with_count(count)
```

Reply:
And, why remove the exception handler? I remember that here was a invalid bz for windows before so we add it to tolerant the exception.

Reply:
> And I recommend here:
> 
> ```
> count = 1000000000 if main_qga_ver < 5 else 10000000
> _read_guest_file_with_count(count)
> ```

this part is editing .thanks recommend it to me .

Reply:
> And, why remove the exception handler? I remember that here was a invalid bz for windows before so we add it to tolerant the exception.

I defined a function named:_read_guest_file_with_count(), move try-exception handler into this function,you can have a review with that.